test/runner/runner_test.dart

```diff
@@ -292,6 +292,82 @@ $_usage""");
     // 15078 is fixed.
   });
 
+  group("retries tests", () {
+    test("and eventually passes for valid tests", () {
+      d
+          .file(
+              "test.dart",
+              """
+          import 'dart:async';
+
+          import 'package:test/test.dart';
+
+          int attempt = 0;
+          void main() {
+            test("failure", () {
+             attempt++;
```

Comment:
Done. Wish there was a way to specify inline code blocks that were formatted better :(